instructor-notes/07_feature-branch.md

```diff
@@ -1,27 +1,60 @@
 
 * Understand the concept of `git` branches and how to make them
-* Understand the concept of `git` merging and how to perform it within a given repository (not across forks)
-  * Merge conflicts will _not_ be covered in depth, but we will mention it
+* Manange more than one branch with independent sets of changes
+* Cherry-pick commits from one branch to another, and revert commits
+
+## Before demonstration
+
+Before the live demo, slides will introduce the follow concepts/commands:
+
+* Exploring project history in GitHub using `scpca-nf`
+  * See past commits, releases and/or tags
+  * `Blame` view (use [`scpca-nf/bin/post_process_sce.R`](https://github.com/AlexsLemonade/scpca-nf/blob/main/bin/post_process_sce.R), a script with three contributors)
+* Introduce feature branches conceptually and why we use them, including informative branch names
+  * Note that more discussion of feature branches will come later in the afternoon during git workflow slides
+* Introduce `git merge` and (why not to use) `git rebase`
+* Introduce `git checkout` (as in `git switch`) and `git branch`
+* Introduce `git stash` and `git cherry-pick`
+
 
 
 ## Activity
 
+### Creating a single branch
 
-* At this time, transition to the "Branch" slides to introduce the concepts of branching and merging.
-* After the slides, interactively demonstrate how to create a new branch, which should be called something like `add-renv`.
+* After the Git part 2 slides, we will demonstrate how to create a new branch, which should be called something like `add-renv`.
 * Trainees should then stage/commit/push the `renv` files to a remote branch, which will be created during the `push`.
+* "Accidentally" include the `environment.yml` file as a second (or third) commit.
 * GitKraken should now show diverged histories between `main` and `add-renv`.
-
 * Instruct trainees to navigate to their remote repositories in the browser to see their new branch pushed.
-* In particular, make sure trainees see the github message along the lines of "This branch 1 commit behind the main branch" inside the `add-renv` branch view.
-* Return to GitKraken to _merge_ this feature branch into their `main` branch, push to `main`, and then delete the remote (and the local if preferred!) `add-renv` branch.
-  * Reiterate point from slides that other types of project workflows might not directly merge branches into `main`
-* Finally, if the local branch was not deleted, ensure trainees end up in their `main` branch locally.
+  * In particular, make sure trainees see the github message along the lines of "This branch is 2 commits ahead of main" inside the `add-renv` branch view.
+  * Point out the pull request button, but do not click it yet.
+
+### Creating the second branch
+
+* Now return to GitKraken to create a separate branch for adding the conda environment.
+* Create a branch off of `main` named `add-fastp`.
+* Make a commit to add the changes to the `download-fastq.sh` script to use `fastp`.
```

Comment:
Should this also include the `-f` version of checking for files, or will you do that later? Not clear to me when later would be.

Reply:
Yes, I meant here that this would include all of the changes to `download-fastq.sh` made in the 06 activity. 

My thought was that when we get to filing the first PR with the `add-renv` branch, you can "forget" that I made those changes and add the `-e` version to "prevent extra downloads." Then you would push one more change before the PR, which I would quickly approve from my iPad. 

Then I pull the merged changes to the `add-fastp` branch when it is time for conflict resolution. 

Does that align with your expectations?

Reply:
Yes. Can we be explicit about that here?

Reply:
It looks like you were in a more recent commit!